File: src/main/java/imaxct/service/IUserService.java
package imaxct.service;

import imaxct.bean.Msg;
import imaxct.domain.User;

import java.util.List;

/**
 * Created by imaxct on 17-4-6.
 */
public interface IUserService {

    Msg<User> login(String username, String password);

    Msg<List<User>> list();
}

File: src/main/java/imaxct/service/IUserService.kt
package imaxct.service

import imaxct.bean.Msg
import imaxct.domain.Course
import imaxct.domain.Selection
import imaxct.domain.User

/**
 * Created by imaxct on 17-4-6.
 */
interface IUserService {

    fun login(stuNo: String, password: String): Msg<User>

    fun list(): Msg<List<User>>

    fun selectCourse(course: Course, user: User): Msg<*>

    fun deSelectCourse(course: Course, user: User): Msg<*>

    fun updateInfo(user: User): Msg<*>

    fun getUserById(id: Int): User

    fun getCourseByUser(user: User): List<Selection>
}
